### ISSUE TITLE
Second Weekend Project - 4Q 2023 - Group 9

### ISSUE THREAD

**Issue by AlexGabrielANDREI** [AlexGabrielANDREI/SolidityBootcampQ42023_week2#2]
Second weekend project - 4Q 2023 - GROUP 9

Contract: https://sepolia.etherscan.io/address/0x5C2D5Da7884f0bcDB452f877c8c0eb8dfC754409

Contract creation by **Gabriel1190**: https://sepolia.etherscan.io/tx/0x6bf4d9c1206a0d8460b3446748136715fea67a8d4bffc2fd5c098bc303ba9a9a

![image](https://github.com/AlexGabrielANDREI/SolidityBootcampQ42023_week2/assets/132738143/c8605706-3eb3-43df-9a69-5e6e570d6852)

**Comment by AlexGabrielANDREI** [comment]
**Gabriel1190** call successfully “vote” with tx:
https://sepolia.etherscan.io/tx/0xfd571c4feac0d4958e084630cb896bd99a313c9af6b9db9b303b99c3bf87e924
![image](https://github.com/AlexGabrielANDREI/SolidityBootcampQ42023_week2/assets/132738143/6c84dc09-22b2-4d72-8137-86a4501ae12b)

**Comment by AlexGabrielANDREI** [comment]
**Gabriel1190** GiveRightToVote to **Stefan Adrian** and **javiermontescarrera** with txs:
https://sepolia.etherscan.io/tx/0x5a09a5aac8188aba487d6c50126499158965d2df5247a8fe325972422d1be367
![image](https://github.com/AlexGabrielANDREI/SolidityBootcampQ42023_week2/assets/132738143/6d8d3a38-3814-43d2-8b69-6794b2836928)


https://sepolia.etherscan.io/tx/0x0f12157c1a876b0f318c8d4b30cc81b432f1b15e53005ecd082700f8229b094c
![image](https://github.com/AlexGabrielANDREI/SolidityBootcampQ42023_week2/assets/132738143/af5ce39b-2c1a-4e71-a77a-261195b6483c)

**Comment by AlexGabrielANDREI** [comment]
**Stefan Adrian** call successfully “vote” with tx:
https://sepolia.etherscan.io/tx/0xb233500973f8baaa4a84575e64d7e40f5eaf42642acd3bae92bf8733ebf44180

![image](https://github.com/AlexGabrielANDREI/SolidityBootcampQ42023_week2/assets/132738143/1b12fbc6-6c43-4fe9-9943-f4bc4dede9b4)

**Comment by AlexGabrielANDREI** [comment]
**javiermontescarrera** call successfully “vote” with tx:
https://sepolia.etherscan.io/tx/0xad262c3d71c9793a585a12b298e686b32c24323a1ec2db7081c5351067749efa

![image](https://github.com/AlexGabrielANDREI/SolidityBootcampQ42023_week2/assets/132738143/23c80e62-45e4-4257-8286-40d94cd17158)

**Comment by AlexGabrielANDREI** [comment]
**javiermontescarrera** created a new script to get the winning proposal, which was merged with the main branch:

![image](https://github.com/AlexGabrielANDREI/SolidityBootcampQ42023_week2/assets/132738143/94863dc4-281b-4962-a7dc-aca2b3f4b67d)

**Comment by AlexGabrielANDREI** [comment]
**Gabriel1190** GiveRightToVote to **jaedongg** with tx:
https://sepolia.etherscan.io/tx/0xb046d838fb1ab95725915076de6b4f10db8a8645b3ff5852ad003605f24b3c89

![image](https://github.com/AlexGabrielANDREI/SolidityBootcampQ42023_week2/assets/132738143/2700a65a-ddcc-4919-88ee-27babe028a8a)

**Comment by AlexGabrielANDREI** [comment]
**javiermontescarrera** forced an exception due to trying to vote more than once:
![image](https://github.com/AlexGabrielANDREI/SolidityBootcampQ42023_week2/assets/132738143/a6ccf4ac-189a-4070-8b54-0b75960d90fb)

**Comment by AlexGabrielANDREI** [comment]
**javiermontescarrera** raised another exception because he tried to give right to vote to **Alice** but it reverted because he isn't the chairperson:

 
![image](https://github.com/AlexGabrielANDREI/SolidityBootcampQ42023_week2/assets/132738143/28e49e2d-d182-4f15-a802-6e50e22baa87)

**Comment by AlexGabrielANDREI** [comment]
**jaedongg** call successfully “vote” with tx:

https://sepolia.etherscan.io/tx/0x5463a7cbb6921fe91a2d347e9f1f1ae93c00c92cae99ec3b31dff74bd8cbb24e

![image](https://github.com/AlexGabrielANDREI/SolidityBootcampQ42023_week2/assets/132738143/407c0d14-958f-492a-bee8-aebbc6c397a1)

**Comment by ASIdevelopment** [comment]
Adrian Stefan added DelegateVote script 
<img width="1003" alt="image" src="https://github.com/AlexGabrielANDREI/SolidityBootcampQ42023_week2/assets/19352559/39436fc2-e80c-45c4-940b-1a7394b18693">

Execution Script:

The provided execution script is written in JavaScript using the ethers library to interact with the Ethereum blockchain.
It takes parameters for the contract address and the address to which the vote should be delegated.
The script configures the Ethereum provider, wallet, and checks the account balance.
It uses Typechain to attach the smart contract and sends a transaction to delegate the vote.
The script monitors the transaction's progress and provides the transaction hash upon completion.
In summary, the delegation part of the contract allows voters to transfer their voting rights to another address, enabling a flexible and secure way to participate in the voting process. The execution script facilitates this delegation process by interacting with the Ethereum blockchain.

**Comment by aliceliu7** [comment]
Alice unsuccessfully voted for option 2. The execution reverted due to Alice not having voting permissions: 
<img width="666" alt="Reverted No Voting Permissions Updated" src="https://github.com/AlexGabrielANDREI/SolidityBootcampQ42023_week2/assets/53191680/c3838d40-aa0b-469a-b604-2006e12627d5">

**Comment by dzgierski19** [comment]
dawidzz unsuccessfully voted for option 1. The execution reverted.
<img width="1225" alt="image" src="https://github.com/AlexGabrielANDREI/SolidityBootcampQ42023_week2/assets/106579123/9e70be66-6baf-4383-9551-d65b1324894c">

**Comment by Gagan921** [comment]
gagan921 unsuccessfully voted for option 2. The execution reverted.
<img width="847" alt="image" src="https://github.com/AlexGabrielANDREI/SolidityBootcampQ42023_week2/assets/63442672/63aa86fb-db39-4d40-9327-731388043e80">
<img width="960" alt="image" src="https://github.com/AlexGabrielANDREI/SolidityBootcampQ42023_week2/assets/63442672/04cef44f-84bb-417e-bcc0-8e3b60e34b2c">

**Comment by Gagan921** [comment]
gagan921 created a new script to get the winning proposal, which was merged with the main branch:
<img width="952" alt="image" src="https://github.com/AlexGabrielANDREI/SolidityBootcampQ42023_week2/assets/63442672/afbd7432-cdc4-4a32-8474-aa4835b66cfa">
